documentation/add_compiler.md

```diff
@@ -0,0 +1,33 @@
+# Add Compiler
+
+The `peadm::add_compiler` plan can be used to add a new compiler to a PE cluster or replace an existing one with new configuration.
+
+## Add a compiler to an existing PE cluster
+
+Prepare to run the plan using a params.json file such as this one:
+
+```json
+{
+  "avail_group_letter": "A",
+  "compiler_host": "pe-xl-compiler-0.lab1.puppet.vm",
+  "dns_alt_names": [ "puppet", "puppet.lab1.puppet.vm" ],
+  "primary_host":  "pe-xl-core-0.lab1.puppet.vm",
+  "primary_postgresql_host": "pe-psql-6251cd-0.us-west1-a.c.slice-cody.internal",
```

Comment:
why do we need to provide the postgresql FQDN/common name? We should be able to query PuppetDB or the node classifier API so users don't need to provide this?

Reply:
@davidsandilands Do you by any chance remember the rationale behind this? We do get the whole cluster configuration from the task `peadm::get_peadm_config` (https://github.com/puppetlabs/puppetlabs-peadm/blob/main/plans/add_compiler.pp#L24)

Reply:
@timidri is this to do with people adding compilers for purely impact analysis or where certain regional zones are setup and they have their own load balancer?

Reply:
@davidsandilands I am not sure what was the rationale. I would suggest making those parameters optional and choose the obvious defaults based on the `peadm::get_peadm_config` output. /cc @cathal41

Reply:
I've made a few change to make a couple of these parameters optional, if anyone would like to take a look at it. https://github.com/puppetlabs/puppetlabs-peadm/pull/468